examples/SpongeRollProblem4.py

```diff
@@ -80,14 +82,14 @@ class Pattern:
     totalRollLength = 20
     lenOpts = [5, 7, 9]
 
-    def __init__(self, name, lengths=None):
+    def __init__(self, name: str, lengths: List[int]) -> None:
```

Comment:
shouldn't we keep the `=None` here?

Reply:
I removed it because the `zip` below on L87 fails to typecheck if `lengths` is `None`. Since this code is presumably working I took that to mean that this is never used that way in practice.